src/rust/lib/graph-editor/src/component/edge.rs

```diff
@@ -44,23 +47,280 @@ const NODE_PADDING       : f32 = node::SHADOW_SIZE;
 const PADDING            : f32 = 4.0;
 const RIGHT_ANGLE        : f32 = std::f32::consts::PI / 2.0;
 
+const SNAP_THRESHOLD     : f32 = 1.2 * LINE_SHAPE_WIDTH;
+
 const INFINITE : f32 = 99999.0;
 
 
 
+// ========================
+// === Edge Shape Trait ===
+// ========================
+
+/// Edge shape defines the common behaviour of the sub-shapes used to create a Edge.
+trait EdgeShape: ensogl::display::Object {
+    fn set_hover_split_offset(&self, offset:Vector2<f32>);
```

Comment:
This needs either documentation or slightly better naming. Is it `set_split_origin`?